modules/hal_nordic/CMakeLists.txt

```diff
@@ -6,3 +6,19 @@ if (CONFIG_NRF_802154_RADIO_DRIVER OR CONFIG_NRF_802154_SERIALIZATION)
 endif (CONFIG_NRF_802154_RADIO_DRIVER OR CONFIG_NRF_802154_SERIALIZATION)
 
 add_subdirectory_ifdef(CONFIG_HAS_NRFX nrfx)
+
+if(CONFIG_NRF_REGTOOL_GENERATE_UICR)
+  list(APPEND nrf_regtool_components GENERATE:UICR)
+endif()
+if(DEFINED nrf_regtool_components)
+  find_package(nrf-regtool 5.0.1
+    COMPONENTS ${nrf_regtool_components}
+    PATHS ${CMAKE_CURRENT_LIST_DIR}/nrf-regtool
+    NO_CMAKE_PATH
+    NO_CMAKE_ENVIRONMENT_PATH
+    NO_SYSTEM_ENVIRONMENT_PATH
+    NO_CMAKE_PACKAGE_REGISTRY
+    NO_CMAKE_SYSTEM_PATH
+    NO_CMAKE_SYSTEM_PACKAGE_REGISTRY
+  )
+endif()
```

Comment:
this now issues a warning if not found, however, the final hex file is in practice unusable if regtool doesn't populate a UICR file. I think we should make this `REQUIRED`.

Reply:
I'd say it's not strictly required, because the user is free to provide their own UICR file separately. You can also turn off `CONFIG_NRF_REGTOOL_GENERATE_UICR`, which would hide the warning.

Reply:
> I'd say it's not strictly required, because the user is free to provide and flash their own UICR file separately. You can also turn off `CONFIG_NRF_REGTOOL_GENERATE_UICR`, which would hide the warning.

but is this a realistic case? My fear is that a warning is easy to ignore, and then, flashing the app will just not work.

Reply:
It's not very realistic, so we could say we don't support it, and make it `REQUIRED`.